fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -920,7 +920,7 @@ public static Expression processComparisonPredicate(ComparisonPredicate comparis
         right = comparisonPredicate.right();
 
         Optional<DataType> commonType = findWiderTypeForTwoForComparison(
-                left.getDataType(), right.getDataType(), false);
+                left.getDataType(), right.getDataType(), true);
```

Comment:
should not change this. because the legacy planner do the same thing for long time